metal/datasource_metal_facility.go

```diff
@@ -33,6 +33,26 @@ func dataSourceMetalFacility() *schema.Resource {
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Computed:    true,
 			},
+			"capacity": {
+				Type:        schema.TypeList,
+				Description: "Optional capacity specification",
+				MaxItems:    1,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"plan": {
+							Type:        schema.TypeString,
+							Description: "Plan which has to be available in selected facility",
+							Required:    true,
+						},
+						"quantity": {
```

Comment:
We don't have to introduce the `level` checks in this PR, but we may want to add them in the future.  

For users that know which facility they want and would like to prevent provisioning if their `var.device_count` won't be met, this will be helpful.

Reply:
- I think level is not part of the capacity query: https://metal.equinix.com/developers/api/capacity/#check-capacity
- I think the levels are useless once you can actually query if a faciltiy (or metro) has *n* devices of a particular plan available.